cpu/atmega_common/Kconfig

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2020 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+config CPU_ARCH_AVR8
+    bool
+    select HAS_ARCH_8BIT
+    select HAS_ARCH_AVR8
+
+config CPU_CORE_AVR
+    bool
+    select CPU_ARCH_AVR8
+
+config CPU_FAMILY_ATMEGA
+    bool
+    select CPU_CORE_AVR
+    select HAS_ATMEGA_PCINT0
+    select HAS_PERIPH_CPUID
+    select HAS_PERIPH_EEPROM
+    select HAS_PERIPH_GPIO
+    select HAS_PERIPH_GPIO_IRQ
+    select HAS_PERIPH_PM
+    select HAS_PERIPH_TIMER_PERIODIC
+    select HAS_PERIPH_WDT
+    select HAS_PUF_SRAM
+
+## Common CPU symbols
+config CPU_ARCH
+    default "avr8" if CPU_ARCH_AVR8
+
+config CPU_CORE
+    default "avr" if CPU_CORE_AVR
+
+config CPU_FAMILY
```

Comment:
This variable is  not documented in `vars.inc.mk`, why `CPU_FAMILY`? `CPU_FAM` is the defined one, as well as the one saved in `makefiles/dependencies_debug.inc.mk`.

Reply:
When I introduced the symbols in  #13715 I though `CPU_FAMILY` was a bit clearer. Kconfig will generate `CONFIG_CPU_FAMILY` from this so if used in the Makefiles we will need to rename them anyway.

Reply:
I know that the usage of these kind of variables is a mess, but adding new variables that are not inline with the current ones is not nice either. I know I'm jumping in very late on this. Could we think of changing all names to either CPU_FAM or CPU_FAMILY, I don't care which one, and this would be done in another PR. But I wanted to comment on it.

Reply:
We could change the name to `CPU_FAM`, but I don't think the variables we introduced are not in line. We added more detail in the categorization. I think it's good taking advantage of this Kconfig effort to organize/complete this information on our platforms.

Reply:
I've taken a second look into how most platforms do this, and other than `CPU` and `CPU_MODEL` other definitions and usage are completely out there, so as long as `CPU_MODEL` and `CPU` are inline with current use I think we are ok. So I would summarize the feedback so far to comments for future PR's:

- replace `CPU_FAM` by `CPU_FAMILY` in our Makefiles
- match help message for `CPU` and `CPU_MODEL` to whats in `vars.inc.mk`

Does that make sense?

Reply:
I seems to me that CPU_FAM is most widely used variable name and CPU_FAMILY is only used by kinetis. Maybe we should just use the former everywhere (including kconfig). This seems like it will be less a pain.

Reply:
I'm OK with moving to `CPU_FAM` for Kconfig. My fear is that, as we are now defining a more detailed classification CPU_FAM may not correspond in Kconfig to the one in Makefiles.

Regarding the help messages, I can change those on a follow-up.

Reply:
> I'm OK with moving to CPU_FAM for Kconfig. My fear is that, as we are now defining a more detailed classification CPU_FAM may not correspond in Kconfig to the one in Makefiles.

That will become an issue anyway whenever we want to model dependencies.

Reply:
I would argue that we should instead keep in sync with what is in the Makefiles, and take every opportunity we can to clean things up. For examples when `CPU_CORE` and `CPU_ARCH` are mixed up.

Reply:
#14246 is trying to unify `CPU_CORE` and `CPU_ARCH`. Regarding `CPU_SERIES` and `CPU_FAMILY`, what do you think about something like the following?
```
   +-----------+
   | CPU_MODEL |
   +-----------+
         |
         v
   +-----------+
   |  CPU_FAM  +---------+
   +-----------+         |
         |               |
         v               v
   +-----------+   +-----------+
   |  CPU_CORE |   |  CPU_LINE |
   +-----------+   +-----------+
         |
         v
   +-----------+
   |  CPU_ARCH |
   +-----------+
```

We rename in Kconfig `CPU_SERIES` to `CPU_FAM`, which would be in line to what is being used now, and `CPU_FAMILY` to `CPU_LINE`.
- `CPU_FAM` would represent a group of MCUs with a particular core and a register map.
- `CPU_LINE` would be a wider group of MCUs from the same vendor and similar characteristics (e.g. `efm32`, `stm32`, `esp`)

Reply:
Thoughts?